src/components/graph-sonification.tsx

```diff
@@ -0,0 +1,174 @@
+      const allItems = allItemsRes.values;
+
+      // Asign pitch and time values to the dataset items
+      // Do not include items that are missing values for either attributes
+      const validItems = pitchAttr && timeAttr
+        ? allItems.filter((item: CodapItem) => item.values[pitchAttr] !== "" && item.values[timeAttr] !== "")
+        : allItems.filter((item: CodapItem) => item.values[timeAttr] !== "");
+
+      const pitchValues = validItems.map((item: CodapItem) => item.values[pitchAttr]);
+      const timeValues = validItems.map((item: CodapItem) => item.values[timeAttr]);
+
+      const minTime = graphDetails.xLowerBound;
+      const maxTime = graphDetails.xUpperBound;
+      const timeRange = maxTime - minTime || 1;
+
+      const minPitch = graphDetails.yLowerBound;
+      const maxPitch = graphDetails.yUpperBound;
+      const pitchRange = maxPitch - minPitch || 1;
+
+      const timeFractions = timeValues.map((t: number) => (t - minTime) / timeRange);
+      const pitchFractions = pitchValues.map((p: number) => (p - minPitch) / pitchRange);
+
+      await Tone.start();
+      Tone.getTransport().cancel();
+      Tone.getTransport().stop();
+      Tone.getTransport().position = 0;
+
+      const totalDuration = 10;
+      const fractionGroups: Record<number, number[]> = {};
+      timeFractions.forEach((frac: number, i: number) => {
+        if (!fractionGroups[frac]) fractionGroups[frac] = [];
+        fractionGroups[frac].push(i);
+      });
+
+      const uniqueFractions = Object.keys(fractionGroups)
+        .map(parseFloat)
+        .sort((a, b) => a - b);
+
+      uniqueFractions.forEach((fraction) => {
+        const offsetSeconds = fraction * totalDuration;
+        const indices = fractionGroups[fraction];
+      
+        Tone.getTransport().scheduleOnce((time) => {
+          indices.forEach((i) => {
+            const pFrac = pitchFractions[i];
+            const freq = 220 + pFrac * (880 - 220);          
+
+            // For each data point, determine the pan value based on the x-axis position.
+            // Normalize the x-axis position to a value between -1 and 1.
+            const panValue = ((timeValues[i] - graphDetails.xLowerBound) / (graphDetails.xUpperBound - graphDetails.xLowerBound)) * 2 - 1;
+            pannerRef.current?.pan.setValueAtTime(panValue, time);
+      
+            synthRef.current?.triggerAttackRelease(freq, "8n", time);
+          });
+        }, offsetSeconds);
+      });
+
+      Tone.getTransport().start();
+      const step = () => {
+        const elapsed = Tone.getTransport().seconds;
+        const fraction = Math.min(elapsed / totalDuration, 1);
+
+        codapInterface.sendRequest({
+          action: "update",
+          resource: `component[${graphDetails.id}].adornment`,
+          values: {
+            type: "Region of Interest",
+            primary: { "position": `${fraction * 100}%`, "extent": 0.05 }
```

Comment:
`"extent": "5%"` ?

Reply:
Using `0.05` is an attempt to make a line that is about 1 pixel in width. We have a better method for determining the necessary value in the next PR.